### PR TITLE
Test if is_nonstr_iter works for numpy arrays

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Add a doctest checking if function `is_nonstr_iter` works for numpy arrays.

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

### CODE REVIEW

File: pygmt/helpers/utils.py [hunk 1]
```diff
@@ -172,6 +172,9 @@ def is_nonstr_iter(value):
     True
     >>> is_nonstr_iter((1, 2, 3))
     True
+    >>> import numpy as np
+    >>> is_nonstr_iter(np.array([1.0, 2.0, 3.0]))
+    True
```
Comment (weiji14) [inline pygmt/helpers/utils.py:177]: Do you want to test with `np.array(["abc", "def", "ghi"])`? :laughing:

By the way, what's the reason for adding this test?

Comment (seisman) [reply pygmt/helpers/utils.py:177]: > Do you want to test with `np.array(["abc", "def", "ghi"])`?

I think yes. It should be `True`.

> By the way, what's the reason for adding this test?

Trying to implement the feature of varying transparency level (#615). `transparency` can be int, float or list/array.

I need to check if `transparency` is an list/array. If yes, then append it to the `extra_arrays`, which will be passed via `GMT_Put_Vector`.

Comment (weiji14) [reply pygmt/helpers/utils.py:177]: Ah I see, looks ok then.

**Review by weiji14 — APPROVED** [review]

**Comment by weiji14** [comment]
By the way, maybe add the `skip-changelog` label since this is a pretty small change.

**Pull request merged**